src/main/java/io/kaitai/struct/IKaitaiEnum.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2015-2024 Kaitai Project: MIT license
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package io.kaitai.struct;
+
+/**
+ * Interface, implemented by every enum class generated by Kaitai Struct Compiler,
+ * which provides assess to the underlying integer value.
```

Comment:
```suggestion
 * Interface implemented by every enum class generated by Kaitai Struct compiler,
 * which provides access to the underlying integer value.
```

I don't really see the point of the "which provides access to the underlying integer value" part, though. It doesn't seem to be a defining characteristic of enums, more like a (obvious) sub-feature. It's probably better to delete it and just leave it at "Interface implemented by every enum class generated by Kaitai Struct compiler."

Reply:
I want to have some words to describe not only _what it that_, but also _why_ it is used. I removed that sentence, but would be glad to hear what can replace it.